frontend/src/components/datasets/ShareDatasetModal.tsx

```diff
@@ -85,34 +85,11 @@ export default function ShareDatasetModal(props: ShareDatasetModalProps) {
 	}, [users, myProfile.email]);
 
 	const onShare = async () => {
-		setShowFailAlert(false);
-		setShowSuccessAlert(false);
 		await setUserRole(datasetId, email, role);
+		// setGroup({ label: "", id: "" });
+		setRole("viewer");
+		setShowSuccessAlert(true);
```

Comment:
I don't quite understand this part `setRole("viewer");`, why hard code to "viewer"?

Reply:
I checked. This is there because if you add a user with another permission, and then you go to add a different user, the default permission will not be 'viewer' but whatever permission you added last. I was thinking that the default any time you go to add someone was supposed to be viewer?

Reply:
I see i understand now. Thanks!